Drivers/Src/MotorController.c

```diff
@@ -46,7 +45,8 @@ void MotorController_Drive(float newVelocity, float motorCurrent){
  * @param   message the buffer in which the info for the CAN message will be stored
  * @return  SUCCESS if a message is read
  */ 
-ErrorStatus MotorController_Read(CANbuff *message){
+ErrorStatus MotorController_Read(CANbuff *message, void *p_arg){
```

Comment:
Is there any particular reason this is taking a void pointer instead of a pointer to a car state? I can't see a reason for this to be generic.